langchain/src/smith/config.ts

```diff
@@ -0,0 +1,248 @@
+
+  /**
+   * Convert the evaluation data into a format that can be used by the evaluator.
+   * By default, we pass the first value of the run.inputs, run.outputs (predictions),
+   * and references (example.outputs)
+   *
+   * @returns The prepared data.
+   */
+  prepareData?: PrepareDataT;
+
+  /**
+   * The language model specification for evaluators that require one.
+   */
+  eval_llm?: string;
+
+  /**
+   * Configuration to load a "CriteriaEvalChain" evaluator,
+   * which prompts an LLM to determine whether the model's
+   * prediction complies with the provided criteria.
+   * @param criteria - The criteria to use for the evaluator.
+   * @param llm - The language model to use for the evaluator.
+   * @returns The configuration for the evaluator.
+   * @example
+   * ```ts
+   * const evalConfig = new RunEvalConfig(
+   *  [new RunEvalConfig.Criteria("helpfulness")],
+   * );
+   * ```
+   * @example
+   * ```ts
+   * const evalConfig = new RunEvalConfig(
+   * [new RunEvalConfig.Criteria(
+   *      { "isCompliant": "Does the submission comply with the requirements of XYZ"
+   *  })],
+   */
+  static Criteria = class implements EvalConfig {
+    evaluatorType: keyof EvaluatorType = "criteria";
+
+    /**
+     * The "criteria" to insert into the prompt template
+     * used for evaluation. See the prompt at
+     * https://smith.langchain.com/hub/langchain-ai/criteria-evaluator
+     * for more information.
+     */
+    criteria?: Criteria | Record<string, string>;
+
+    /**
+     * The feedback (or metric) name to use for the logged
+     * evaluation results. If none provided, we default to
+     * the evaluationName.
+     */
+    feedbackKey?: string;
+
+    /**
+     * The language model to use as the evaluator.
+     */
+    llm?: BaseLanguageModel;
+
+    constructor(props: {
+      criteria?: Criteria | Record<string, string>;
+      llm?: BaseLanguageModel;
+      feedbackKey?: string;
+    }) {
+      this.criteria = props.criteria;
+      this.llm = props.llm;
+      this.feedbackKey = props.feedbackKey;
+    }
+  };
+
+  /**
+   * Configuration to load a "LabeledCriteriaEvalChain" evaluator,
+   * which prompts an LLM to determine whether the model's
+   * prediction complies with the provided criteria and also
+   * provides a "ground truth" label for the evaluator to incorporate
+   * in its evaluation.
+   * @param criteria - The criteria to use for the evaluator.
+   * @param llm - The language model to use for the evaluator.
+   * @returns The configuration for the evaluator.
+   * @example
+   * ```ts
+   * const evalConfig = new RunEvalConfig(
+   *  [new RunEvalConfig.LabeledCriteria("correctness")],
+   * );
+   * ```
+   * @example
+   * ```ts
+   * const evalConfig = new RunEvalConfig(
+   * [new RunEvalConfig.Criteria(
+   *      { "mentionsAllFacts": "Does the include all facts provided in the reference?"
+   *  })],
+   */
+  static LabeledCriteria = class implements EvalConfig {
```

Comment:
My gut is these don't need to be classes - also, no need to nest them?

Reply:
Ya - don't feel terribly strongly about nesting, but that makes it similar to python. I  was running into issues defining nested types so could also just have flat types instead of nested classes I believe